### DISCUSSION TITLE
Filament crashes for a vector asset

### DISCUSSION THREAD

**Discussion by soruppuAswin** [google/filament#5971] [Q&A]
Hi,

I am doing the following to create a MaterialInstance to load the plane texture. If we make use of a vector asset for the same, `loadTexture` fails to get the bitmap and the app crashes at the Texture builder. Can anyone help me how could I address this issue? We are only allowed to use a vector asset. 

```
        return textureMaterial
            .createInstance()
            .also { materialInstance ->
                materialInstance.setParameter(
                    "texture",
                    com.google.android.filament.textured.loadTexture(
                        engine,
                        context.resources,
                        **PNG DRAWABLE ASSET**
                        TextureType.COLOR
                    ),
                    TextureSampler().also { it.anisotropy = 8.0f },
                )

                materialInstance.setParameter("alpha", 1f)
            }
```

**Comment by romainguy** [comment]
Vector assets are not `Bitmap` instances. You need to draw your vector drawable into a `Bitmap` first.

**Reply by soruppuAswin** [reply]
Got you. By the way, once I get a bitmap, how can we associate the same to a Texture/MaterialInstance?

**Reply by romainguy** [reply]
I'm not sure I understand? You create a `Texture` object from the `Bitmap` and then you can use that `Texture` as a parameter to a `MaterialInstance`.
